tests/unit/test_interface.py

```diff
@@ -1,8 +1,78 @@
+from pathlib import Path
+
 import pytest
+from packaging.version import parse
+
+from pseudopeople.exceptions import DataSourceError
+from pseudopeople.interface import (
+    _get_data_changelog_version,
+    validate_source_compatibility,
+)
+
+
+# TODO [MIC-4546]: stop hardcoding the data version number
+@pytest.fixture()
+def simulated_data_changelog_path(tmpdir_factory):
+    """Returns the path to where simualted data would live alongside their
+    respective CHANGELOG.rst. This fixture creates the changelog but does
+    not actually contain any simulated data.
+    """
+    tmp_path = str(tmpdir_factory.getbasetemp())
+    filepath = f"{tmp_path}/CHANGELOG.rst"
+    changelog_content = (
+        "**1.4.2 - some other date**"
+        "\n\n"
+        " - Did some other things"
+        "\n\n"
+        "**1.4.1 - some date**"
+        "\n\n"
+        " - Did some things"
+        "\n"
+    )
+    with open(filepath, "w") as file:
+        file.write(changelog_content)
+
+    # Create a dir with no changelog
+    tmpdir_factory.mktemp("no_changelog_dir")
+
+    return Path(tmp_path)
+
+
+def test__get_data_changelog_version(simulated_data_changelog_path):
+    """Test that the data version is extracted from the CHANGELOG correctly"""
+    assert _get_data_changelog_version(simulated_data_changelog_path / "CHANGELOG.rst")
```

Comment:
You're not actually asserting that the version equals anything in particular, just that it doesn't error and isn't false.

Reply:
Haha oops